idl/forward_request.idl.hh

```diff
@@ -1,15 +1,31 @@
+namespace db {
+namespace functions {
+class function_name {
+    sstring keyspace;
+    sstring name;
+};
+}
+}
 namespace query {
 struct forward_request {
+    struct aggregation_info {
+        db::functions::function_name name;
+        std::vector<sstring> column_names;
+    };
     enum class reduction_type : uint8_t {
         count,
+        aggregate
     };
-    std::vector<query::forward_request::reduction_type> reduction_types;
+
+    std::vector<query::forward_request::reduction_type> reductions;
 
     query::read_command cmd;
     dht::partition_range_vector pr;
 
     db::consistency_level cl;
     lowres_clock::time_point timeout;
+
+    std::optional<std::vector<query::forward_request::aggregation_info>> aggregation_infos;
```

Comment:
you didn't mark that with a version from which it's supported, like it's done here: https://github.com/scylladb/scylla/blob/6c818f86256ba8c098fca3526e0f25232093f0b6/idl/frozen_schema.idl.hh#L21 . That's why you're getting an underflow. Mark the code with `[[5.1]]` and retest - you should be fine.

Reply:
And then, when receiving from the network, this parameter will automatically be wrapped in `rpc::optional`, because it's marked as only existing from a specific release.

Reply:
I think the marking is just informational (misinformational in this case) and doesn't affect the generated code.

Reply:
@avikivity maybe it used to be only informational, but I see in the IDL compiler code, that it wraps the generated attribute in rpc::optional:
https://github.com/scylladb/scylla/blob/630959bb77873c60b873bdb18afa367a18fc6048/idl-compiler.py#L374-L375

Reply:
88f9f2e9d0

Reply:
The attribute was enough to fix this.